src/client.rs

```diff
@@ -25,11 +27,15 @@ impl Client {
             if let Some(handler) = handler {
                 executor.spawn(Box::new(handler)).unwrap();
             }
-
-            Ok(Client { handle })
+            Ok(Client { handle,  server_address })
         })
     }
 
+    pub fn server_address(&self) -> SocketAddr {
+        let addr: SocketAddr = self.server_address.clone();
```

Comment:
We can avoid the clone here and just do:

```rust
pub fn server_address(&self) -> &SocketAddr {
    &self.server_address
}
```

Reply:
that's way nicer -- thx!